src/Apps/Order/Routes/Shipping2/Components/SavedAddresses2.tsx

```diff
@@ -54,28 +54,27 @@ export const SavedAddresses2: FC<SavedAddressesProps> = props => {
 
   const addressSavedToOrderID = savedAddressOnOrder?.internalID
 
-  // TODO: Make sure this can't create an infinite loop if submitting fails
+  // Automatically select (save) best available address ID if it isn't present
   useEffect(() => {
-    // Automatically select (save) best available address ID if it isn't present
-    const automaticallySelectBestAddress = async () => {
-      if (
-        props.active &&
-        !shippingContext.state.isPerformingOperation &&
-        !formikContext.isSubmitting &&
-        !shippingContext.state.selectedSavedAddressID &&
-        addressList.length > 0
-      ) {
-        const bestAddress = getBestAvailableAddress(
-          addressList,
-          addressSavedToOrderID,
-          shippingContext.orderData.availableShippingCountries
-        )
-        if (bestAddress) {
-          selectAndSubmitAddress(bestAddress)
-        }
-      }
+    const activeAndNoAddressSaved =
+      props.active &&
+      !shippingContext.state.isPerformingOperation &&
+      !formikContext.isSubmitting &&
+      !shippingContext.state.selectedSavedAddressID &&
+      addressList.length > 0
+
+    if (!activeAndNoAddressSaved) {
+      return
+    }
+
+    const bestAddress = getBestAvailableAddress(
+      addressList,
+      addressSavedToOrderID,
+      shippingContext.orderData.availableShippingCountries
+    )
+    if (bestAddress) {
+      selectAndSubmitAddress(bestAddress)
     }
```

Comment:
removed the named function for clarity from inside here but the logic remains the same.